### DISCUSSION TITLE
Tiles explained for use with UpdateDisplayArea

### DISCUSSION THREAD

**Discussion by Blokhuizen** [olikraus/u8g2#1367] [Q&A]
I'm struggling with the UpdateDisplayArea and hope you can help.

The project I'm working on is a alarm-clock.
To set the alarm-time, I'm using a rotary button (which is working fine).
When you click the button, the Hour number has to blink. Turning the button will in- or decrease the number, entering the button again the Minutes has to blink. 
When the time has been set, the AM/PM has to blink.

![foto_display](https://user-images.githubusercontent.com/51677431/103035619-fd62f100-4567-11eb-8456-d44d5c65f159.jpg)

See code below, (it's just testing code !!) for updating the area's.
I've managed to determine the Tile positions and format for the Hour and Minute, but I'm unable to find the correct settings for the AM/PM.

How can I calculate the proper Tile positions and dimensions (in Tiles) for the data on the screen?
And, maybe there is a better way to create blinking "texts" ?

`void DrawTime(bool SetHour, bool SetMinute,bool SetAMPM)
{

    // Drawing text in White
	
    u8g2.setFontMode(1);
    u8g2.setFont(u8g2_font_logisoso34_tf);
    u8g2.setDrawColor(1); //1 = white
    u8g2.drawStr( 1, 57,  1:23);

    u8g2.setFont(u8g2_font_5x7_tf);
    u8g2.drawStr( 119, 64, "AM");
    u8g2.sendBuffer();
	
	
	// Create buffer with same text in Black
	// only specific area will be updated to create blinking effect
	
	
    u8g2.setFont(u8g2_font_logisoso34_tf);
    u8g2.setDrawColor(0); //1 = black
    u8g2.drawStr( 1, 57,  1:23);

    u8g2.setFont(u8g2_font_5x7_tf);
    u8g2.drawStr( 119, 64, "AM");
    u8g2.sendBuffer();
	
    
        
    if (SetHour == true)
    {
      u8g2.updateDisplayArea(8, 2.5, 7.75, 6.25); // Hour
    }
    else if (SetMinute == true)
    {
        u8g2.updateDisplayArea(7, 2.5, 6.25, 6.25); // Minutes 
    }
    else if (SetAMPM == true)
    {
      u8g2.updateDisplayArea(???, ???, ???, ???); // AM/PM 
    }
    delay(200);

}
`

**Comment by olikraus** [comment]
I do not understand the code. Executing updateDisplayArea after sendBuffer does not make sense.
In general you could solve the problem without updateDisplayArea, just repaint everything and then call sendBuffer().

If you need a blinking minutes, then just draw everything, wait for some time and again draw everything without minutes, again wait some time and then again redraw everything.
Suggestion is to create a draw procedure with several flags, like this:
``` C++
void draw(int isMinutes, int isHour, int isColon )
{
  drawUpperLeftBell();
  drawAlarm();
  if ( isHour ) drawHour();
  if ( isColon ) drawColon();
  if ( isMonutes) drawMinutes();
  ...
  sendBuffer();
]
```
Now you can blink minutes with:
``` C++
for(;;) 
{
draw(1,1,1);
delay(100);
draw(0,1,1);
delay(100)
}
```
For blinking hours, just change the flag:
``` C++
for(;;) 
{
draw(1,1,1);
delay(100);
draw(1,0,1);
delay(100)
}
```
